site/content/en/docs/concepts/cluster_queue.md

```diff
@@ -422,12 +422,52 @@ The fields above do the following:
   - `LowerOrNewerEqualPriority`: only preempt Workloads in the ClusterQueue that either have a lower priority than the pending workload or equal priority and are newer than the pending workload.
 
 Note that an incoming Workload can preempt Workloads both within the
-ClusterQueue and the cohort. Kueue implements heuristics to preempt as few
-Workloads as possible, preferring Workloads with these characteristics:
+ClusterQueue and the cohort.
 
-- Workloads belonging to ClusterQueues that are borrowing quota.
-- Workloads with the lowest priority.
-- Workloads that have been admitted more recently.
+Kueue implements heuristics to preempt as few Workloads as possible.
+Below we present a more detailed description of the algorithm.
+
+### Preemption Algorithm overview
+
+An incoming Workload, which does not fit within the free capacity, is eligible
+for preemptions, within a flavor of the target queue, when one of the following
+is true:
+- the requests of the Workload are below the flavor's nominal quota, or
+- `borrowWithinCohort` is enabled.
+
+#### Candidates
+
+The list of preemption candidates is compiled from Workloads within the Cluster
+Queue satisfying the `withinClusterQueue` policy, and Workloads within the
+cohort which satisfy the `reclaimWithinCohort` policy.
+
+The list of candidates is sorted based on the following preference checks for
+tie-breaking:
+- Workloads from other borrowing queues in the cohort,
+- Workloads with the lowest priority,
+- Workloads which got admitted the most recently.
+
+#### Targets
+
+The algorithm qualifies the candidates as targets using the following heuristics:
+
+1. If all candidates belong to the target queue, then Kueue greedily
+qualifies candidates until the incoming Workload can fit, allowing for borrowing.
```

Comment:
instead of `allowing borrowing`, maybe we can be more explicit.

And then you can refer to "allowing/disallowing borrowing" freely

```
allowing the usage of the ClusterQueue to be above the nominal quota, up to the `borrowingLimit`.
This is called "borrowing".
```

Reply:
Done along the lines. PTAL.